adobe/react-spectrum
# fix(#2370): Fix focus management and accessibility when toggling Hamburger menu button

## 🙋 Documentation Request

I noticed a couple small accessibility problems with Hamburger menu in documentation site caused by an improper CSS selector in the code.

1. The querySelector here selects the container element for the Hamburger menu button: https://github.com/adobe/react-spectrum/blob/11596864232ede67642f304e175a678eafcbb176/packages/dev/docs/src/client.js#L95
2. `aria-pressed` should be added to the button within the container element rather than the container element itself, and we should use the `setIsPressed` method to update the state rather than DOM manipulation:
https://github.com/adobe/react-spectrum/blob/11596864232ede67642f304e175a678eafcbb176/packages/dev/docs/src/client.js#L100
3. We should focus the Hamburger menu button rather than the container element here:  https://github.com/adobe/react-spectrum/blob/11596864232ede67642f304e175a678eafcbb176/packages/dev/docs/src/client.js#L103

I have a more robust fix for these issues using `useRef`.

## 🧢 Your Company/Team

Adobe/Accessibility


Closes #2370 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 2370](https://github.com/adobe/react-spectrum/issue/2370).

## 📝 Test Instructions:

1. Open https://react-spectrum.adobe.com/index.html or https://reactspectrum.blob.core.windows.net/reactspectrum/79ecd4fb9d247dc4e09104bc66d65629061354e6/docs/index.html
2. Adjust the browser width so that the Hamburger menu button displays and the SideNav hides.
3. Press the Hamburger menu button to expand the SideNav overlay.
4. Press Tab key to move focus into the SideNav overlay.
5. Press Esc key to close the menu.
6. Keyboard focus should be restored to the Hamburger menu button when the SideNav closes. 

Current behavior:
https://react-spectrum.adobe.com/index.html

Expected behavior: https://reactspectrum.blob.core.windows.net/reactspectrum/79ecd4fb9d247dc4e09104bc66d65629061354e6/docs/index.html

## 🧢 Your Project:

Adobe/Accessibility
